draft-ietf-tls-esni.md

```diff
@@ -339,6 +323,40 @@ Clients MUST parse the extension list and check for unsupported mandatory
 extensions. If an unsupported mandatory extension is present, clients MUST
 ignore the `ECHConfig`.
 
+This specification introduces the following mandatory ECHConfig extension,
+"public_name":
+
+~~~
+   enum {
+       public_name(0x9200), (65535)
+   } ExtensionType;
+~~~
+
+The payload of this extension has the following structure:
+
+~~~
+   struct {
+      opaque public_name<1..255>;
+   } PublicName;
+~~~
+
+This is the non-empty identity of the client-facing server, i.e., the entity
+trusted to update the ECH configuration. This is used to correct misconfigured
+clients, as described in {{handle-server-response}}. This value MUST NOT begin
+or end with an ASCII dot and MUST be parsable as a dot-separated sequence of LDH
+labels, as defined in {{!RFC5890}}, Section 2.3.1. Clients MUST ignore any
+`ECHConfig` structure whose `public_name` does not meet these criteria. Note
+that this criteria incidentally permits a IPv4 addresses in standard
+dotted-decimal or other non-standard notations such as octal and hexadecimal
+(see {{!RFC3986}}, Section 7.4). Clients MUST ignore `ECHConfig` structures
+whose "public_name" payloads contain such addresses.
+
+The value of the "public_name" extension is hereafter referred to as the public
+identity. ECHConfig MUST contain this extension. Future extensions MAY override
```

Comment:
Perhaps: "ECHConfig MUST contain this extension, and clients MUST ignore ECHConfigs missing the extension."